spec/rubocop/cop/lint/next_without_accumulator_spec.rb

```diff
@@ -0,0 +1,43 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Lint::NextWithoutAccumulator do
+  subject(:cop) { described_class.new }
+
+  context 'given a reduce block' do
+    it 'registers an offense for a bare next' do
+      inspect_source(cop, '(1..4).reduce(0) {|a, i| next if i.odd?; a + i }')
+      expect(cop.highlights).to eq(['next'])
```

Comment:
I'd still add here a check there's one offence detected just to be on the safe side.